lib/hint.js

```diff
@@ -0,0 +1,231 @@
+   * @memberof IndexHint
+   */
+  constructor(value, indexHintType, useType){
+    super(value, HINT_TYPE.INDEX_HINT);
+    if (indexHintType && !Object.keys(INDEX_HINT_TYPE).includes(indexHintType)) {
+      throw new LeoricHintError(indexHintType);
+    }
+    if (useType && !Object.values(INDEX_HINT_USE_TYPE).includes(useType)) {
+      throw new LeoricHintError(useType);
+    }
+    this.#indexHintType = indexHintType || INDEX_HINT_TYPE.USE;
+    this.#useType = useType;
+  }
+
+  set indexHintType(type) {
+    if (!type) return;
+    if (Object.keys(INDEX_HINT_TYPE).includes(type.toUpperCase())) {
+      this.#indexHintType = type;
+    }
+    throw new LeoricHintError(type);
+  }
+
+  get indexHintType() {
+    return this.#indexHintType;
+  }
+
+  get useType() {
+    return this.#useType;
+  }
+
+  set useType(type) {
+    if (!type) return;
+    if (Object.values(INDEX_HINT_USE_TYPE).includes(type.toUpperCase())) {
+      this.#useType = type;
+    }
+    throw new LeoricHintError(type);
+  }
+
+  toSqlString() {
+    if (!this.value || !this.indexHintType) return '';
+    const chunks = [this.indexHintType, 'INDEX'];
+    if (this.#useType) {
+      chunks.push(`FOR ${this.useType}`);
+    }
+    if (typeof this.value === 'string') {
+      chunks.push(`(${this.value})`);
+      return chunks.join(' ');
+    }
+    if (Array.isArray(this.value) && this.value.some(item => typeof item === "string")) {
+      chunks.push(`(${this.value.join(',')})`);
+      return chunks.join(' ');
+    }
+    return '';
+  }
+
+  /**
+   *
+   * @param {IndexHint} hint
+   * @returns {boolean}
+   * @memberof IndexHint
+   */
+  isEqual(hint) {
+    if (!hint) return false;
+    return hint.type === this.type && this.value === hint.value && this.indexHintType === hint.indexHintType && this.useType === hint.useType;
+  }
+
+  static copy(hint) {
+    return new IndexHint(hint.value, hint.indexHintType, hint.useType);
+  }
+
+  // static
+  /**
+   *
+   *
+   * @static
+   * @param {IndexHint} hints
+   * @returns {Array<IndexHint>}
+   * @memberof IndexHint
+   */
+  static merge(hints) {
+    if (!hints || !hints.length) return hints;
+    const grouped = {};
+    for (const hint of hints) {
+      const key = `${hint.indexHintType}_${hint.useType || ''}`;
+      if (Object.keys(grouped).includes(key)) {
+        grouped[key].push(hint);
+      } else {
+        grouped[key] = [hint];
+      }
+    }
+
+    const groupArr = [];
+    for (const key in grouped) {
+      let newIndexHint;
+      if (grouped[key].length === 1) {
+        const hint = grouped[key][0];
+        newIndexHint = IndexHint.copy(hint);
+      } else {
+        const sampleHint = grouped[key][0];
+        newIndexHint = new IndexHint(grouped[key].map(item => item.value), sampleHint.indexHintType, sampleHint.useType);
+      }
+      groupArr.push(newIndexHint);
+    }
+
+    return groupArr;
+  }
+}
+
+// simulate interface
+function buildHint(obj) {
```

Comment:
build hint with plain object

Reply:
a better place to put this method is to define it as a static method of `Hint`

```js
class Hint {
  static build() {}
}
```